src/wrapt/_wrappers.c

```diff
@@ -8,6 +8,26 @@
 #define PyVarObject_HEAD_INIT(type, size) PyObject_HEAD_INIT(type) size,
 #endif
 
+/* Heap types in Python < 3.9 don't support __weaklistoffset__ and
+ * __dictoffset__ in PyMemberDef.
+ */
+#if PY_VERSION_HEX >= 0x03090000
+#define WRAPT_HEAPTYPE_SUPPORT_OFFSET 1
+#endif
+
+/* Instanes of heap types in Python < 3.8 don't hold strong ref to their
```

Comment:
Fairly sure this is not the case:
```python
Python 3.6.7 (default, Oct 22 2018, 11:32:17)
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> class C: pass
...
>>> import sys
>>> sys.getrefcount(C)
5
>>> x = [C() for _ in range(100)]
>>> sys.getrefcount(C)
105
```

In code:
https://github.com/python/cpython/blob/v3.6.15/Objects/typeobject.c#L946
https://github.com/python/cpython/blob/v3.6.15/Objects/typeobject.c#L1228